### PR TITLE
Make remote translog store path consistent to remote segment store

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
This PR updates remote translog store path to be consistent with segment.

#### Remote Segment Store Path 
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/segments/data/`
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/segments/metadata/`

#### Previous Translog paths
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/<primaryterm>/(*.tlog|*.ckp)`
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/metadata`

#### New Translog paths
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/data/<primaryterm>/(*.tlog|*.ckp)`
- `s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/metadata`

### Related Issues
Addresses #1 in https://github.com/opensearch-project/OpenSearch/issues/3589#issuecomment-1534329922

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java [hunk 1]
```diff
@@ -68,7 +69,7 @@ public TranslogTransferManager(
     ) {
         this.shardId = shardId;
         this.transferService = transferService;
-        this.remoteBaseTransferPath = remoteBaseTransferPath;
+        this.remoteBaseTransferPath = remoteBaseTransferPath.add(DATA_DIR);
```
Comment (gbbafna) [inline server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java:72]: this no longer remains base , but becomes `remoteDataTransferPath` rather.

File: server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java [hunk 2]
```diff
@@ -68,7 +69,7 @@ public TranslogTransferManager(
...
         this.remoteMetadataTransferPath = remoteBaseTransferPath.add(METADATA_DIR);
```
Comment (gbbafna) [inline server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java:73]: this looks wrong as we are appending `DATA_DIR` as well in previous line.

**Review by BhumikaSaini-Amazon — COMMENTED** [review]
Thanks for adding this!

To confirm, would the new translog paths be:
```
s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/translog/<shardid>/data/<primaryterm>/(*.tlog|*.ckp)
s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/translog/<shardid>/metadata
```

OR

```
s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/data/<primaryterm>/(*.tlog|*.ckp)
s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/metadata
```

i.e. what is the sequence of `<shardid>` and `translog`?

File: server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java (hunk 1 above)
Comment (linuxpi) [reply server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java:72]: Sure, i'll update the name

File: server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java (hunk 2 above)
Comment (linuxpi) [reply server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java:73]: we appended in instance variable, this is the parameter to constructor

Comment (gbbafna) [reply server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java:73]: oh got it . renaming should clear the confusion now.

**Comment by linuxpi** [comment]
> Thanks for adding this!
> 
> To confirm, would the new translog paths be:
> 
> ```
> s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/translog/<shardid>/data/<primaryterm>/(*.tlog|*.ckp)
> s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/translog/<shardid>/metadata
> ```
> 
> OR
> 
> ```
> s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/data/<primaryterm>/(*.tlog|*.ckp)
> s3://remote-store-perf-test/cluster-1/zqQpla0QS_mFK3Q42FbQ0A/<shardid>/translog/metadata
> ```
> 
> i.e. what is the sequence of `<shardid>` and `translog`?

Thanks for catching this! I updated the description, i mistakenly put shardid after translog keyword

**Review by gbbafna — APPROVED** [review]

**Pull request merged**